File: typescript/examples/getting-started/01a-using-schemas.ts
// Using schemas — recommended approach
// Define your schema once with Zod, get type inference and JSON Schema for the API.
import { z } from "zod";
import { Opper } from "../../src/index.js";

const opper = new Opper();

const result = await opper.call("sdk-test-extract-entities", {
  input_schema: z.object({ text: z.string() }),
  output_schema: z.object({
    people: z.array(z.object({ name: z.string(), role: z.string().optional() })),
    locations: z.array(z.string()),
  }),
  input: {
    text: "Marie Curie conducted groundbreaking research on radioactivity in Paris. She was the first woman to win a Nobel Prize. Her husband was Pierre Curie.",
  },
  model: "anthropic/claude-sonnet-4.6",
});
console.log("Full result object:", JSON.stringify(result, null, 2));
console.log("People:", result.data.people); // typed!
console.log("Locations:", result.data.locations); // typed!

File: python/examples/getting-started/01a_using_schemas.py
# Using schemas — recommended approach with Pydantic
from pydantic import BaseModel

from opperai import Opper


class Person(BaseModel):
    name: str
    role: str | None = None


class Entities(BaseModel):
    people: list[Person]
    locations: list[str]


opper = Opper()

result = opper.call(
    "sdk-test-extract-entities",
    input={
        "text": "Marie Curie conducted groundbreaking research on radioactivity in Paris. "
        "She was the first woman to win a Nobel Prize. Her husband was Pierre Curie.",
    },
    output_schema=Entities,
    model="vertexai/gemini-2.5-flash",
)

print("Full result:", result)
print("People:", [person.name for person in result.data.people])
print("Locations:", result.data.locations)
